api/ai/priority_algorithm/priority_algorithm.py

```diff
@@ -78,7 +78,9 @@ def generate(self, students: List[Student]) -> TeamSet:
         ):
             new_team_sets: List[PriorityTeamSet] = []
             for team_set in team_sets:
-                new_team_sets += self.mutate(team_set)
+                new_team_sets += self.algorithm_config.MUTATIONS.generate_mutations(
```

Comment:
Nothing wrong, but might still be useful to leave the self.mutate() function intact for now.

Like just call self.mutate() and put this line of code in there. Idk why but I have a feeling we should keep this unneeded layer of abstraction in case of future work